### PR TITLE
Extending taggit.admin.TagAdmin to include document counts

### PR DESCRIPTION
**Pull request by owenduffymassey**
For [issue#909](https://github.com/Princeton-CDH/geniza/issues/909).  Let me know if there are any additional issues! Thanks for the help.

### CODE REVIEW

**Comment by owenduffymassey** [comment]
Here is how my local setup runs tags with the document counts.
Screen Shot 2022-07-21 at 4 24 19 PM

File: geniza/corpus/admin.py [hunk 1]
```diff
@@ -592,3 +594,32 @@ def save_model(self, request, obj, form, change):
...
+    list_display = ("name", "slug", "document_count")
+
+    def get_queryset(self, request):
+        # The annotations we use for document count on the list view
+        # make the search too slow for autocomplete.
+        # Reset to original, unannotated queryset *only* for autocomplete
+        qs = super().get_queryset(request)
+        if request and request.path == "/admin/autocomplete/":
+            # return without annotations
+            return qs
```
Comment (rlskoeser) [inline geniza/corpus/admin.py:613]: Did you test if taggit uses admin querysets for autocompletes? I think taggit probably custom autocomplete views, so these lines are unneeded.

File: geniza/corpus/admin.py [hunk 2]
```diff
@@ -592,3 +594,32 @@ def save_model(self, request, obj, form, change):
...
+        # otherwise, annotate with counts
+        qs = qs.annotate(
+            # taggit_taggeditem_items is how to reference the taggeditems for tags
+            document_count=Count("taggit_taggeditem_items", distinct=True),
```
Comment (rlskoeser) [inline geniza/corpus/admin.py:617]: Technically this is counting any tagged item, not just documents; we're only currently using tags for documents but I expect that to change in the future; so I suggest `item_count` for the field and `Count` for the label

**Review by rlskoeser — COMMENTED** [review]
Implementation looks good.

Please make the following changes:
- [ ] move the custom tag admin code to `geniza.common.admin` since it isn't specific to the corpus app (and we already have a lot of code in corpus admin)
- [ ] remove unneeded autocomplete exception in `get_queryset`
- [ ] write unit tests for the new methods (get queryset & document count); can probably adapt from similar tests in the corpus app

**Comment by owenduffymassey** [comment]
For the custom tag admin test, is it ok verify the item count number in the query set test? Also, what file should this test be in?

**Comment by rlskoeser** [comment]
Hi @owenduffymassey 

> For the custom tag admin test, is it ok verify the item count number in the query set test? 

We run code coverage reports, so both of the methods you wrote should be tested (even though there is overlap). But that's the basic idea — create some test tags, call the queryset method, and confirm the counts are set as expected.

> Also, what file should this test be in?

Tests are organized by app, so these should be added to `geniza.common.tests`

**Comment by owenduffymassey** [comment]
Ok great! Thanks for the help

**Review by rlskoeser — APPROVED** [review]

**Review by rlskoeser — CHANGES_REQUESTED** [review]
Ah, my apologies! I somehow missed that you opened your PR against `main`. We use git flow, so the PR needs to be to `develop`. I edited the PR to change the target but now it's including a bunch of irrelevant changes.

Would you please see if you can easily rebase your branch against current `develop` code? Or if that doesn't work, create a new branch based on `develop` and cherry-pick the commits from this branch.

Once that is resolved, this is ready to merge in and then I can deliver for acceptance testing.

**Comment by owenduffymassey** [comment]
I just rebased my code against develop, should I continue by pushing with [`git push --force`](https://riptutorial.com/git/example/4357/pushing-after-a-rebase)?

**Comment by rlskoeser** [comment]
> I just rebased my code against develop, should I continue by pushing with [`git push --force`](https://riptutorial.com/git/example/4357/pushing-after-a-rebase)?

yes, go ahead!

**Comment by owenduffymassey** [comment]
I just pushed the new pull request against develop! For some reason the rebase wasn't working well for me.

**Comment by rlskoeser** [comment]
cherry-picked into a fresh branch from develop and merged in via PR #981

**Pull request closed without merging**